File: source python/Organisms/Animals/Fox.py
from Application.Organisms.Animals.Animal import Animal
from Application.Utilities import Colors


class Fox(Animal):
    STRENGTH = 3
    INITIATIVE = 7

    def __init__(self, position, world, **kwargs):
        super().__init__(Fox.STRENGTH, Fox.INITIATIVE, position, world, **kwargs)

    def Act(self):
        class FoxRequirements:
            def __init__(self, strength):
                self.__strength = strength

            def IsOk(self, organism):
                if organism is None:
                    return True
                return organism.GetStrength() <= self.__strength

        self._MoveInNeighbourhood(FoxRequirements(self.GetStrength()))

    @staticmethod
    def __str__():
        return "Fox"

    def GetColor(self):
        return Colors.ORANGE

    def GetLetter(self):
        return "L"

File: source java/organism/animals/Fox.java
package organism.animals;

import Interfaces.Requirements;
import java.awt.Color;
import organism.Organism;
import organism.manager.Organisms;
import utilities.position.Position;
import utilities.savingsystem.Loader;
import virtualworld.World;

public class Fox extends Animal
{
    private static final int STRENGTH = 3;
    private static final int INITIATIVE = 7;
    
    public Fox(Position position, World world) 
    {
        super(STRENGTH, INITIATIVE, position, world);
    }
    
    public Fox(Loader loader, World world, Position position)
    {
        super(loader, world, position);
    }
    
    @Override
    public void Act()
    {
        MoveInNeighbourhood( new Requirements()
        {
            @Override
            public boolean IsOk(Organism organism)
            {
                if(organism == null)
                    return true;
                return organism.GetStrength() > GetStrength();
            }
        });
    }
    
    @Override
    public Organisms GetEnum() 
    {
        return Organisms.FOX;
    }
    
    @Override
    public Color GetColor()
    {
        return Color.orange;
    }
    
    @Override
    public String toString()
    {
        return "Fox";
    }

}
